src/main/java/me/mrCookieSlime/Slimefun/Objects/SlimefunItem/abstractItems/AContainer.java

```diff
@@ -422,6 +422,8 @@ protected MachineRecipe findNextRecipe(BlockMenu inv) {
             for (ItemStack input : recipe.getInput()) {
                 for (int slot : getInputSlots()) {
                     if (SlimefunUtils.isItemSimilar(inventory.get(slot), input, true)) {
+                        if (found.containsKey(slot)) continue; //skip scanned slots
```

Comment:
You should invert this check and merge it with the surrounding of statement, have it be the first boolean expression